lib/ansible/plugins/inventory/onesixtyone.py

```diff
@@ -0,0 +1,244 @@
+# Copyright (c) 2019, Victor da Costa <victorockeiro@gmail.com>
+# Copyright (c) 2019, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: onesixtyone
+    plugin_type: inventory
+    version_added: "2.8"
+    short_description: Uses onesixtyone to find hosts to target
+    description:
+        - Uses any YAML file which ends with onesixtyone.yml or onesixtyone.yaml as a configuration file.
+    extends_documentation_fragment:
+      - constructed
+      - inventory_cache
+    requirements:
+      - onesixtyone CLI installed
+      - python-gevent installed
+      - netaddr python library installed
+    options:
+        addresses:
+            description: List of CIDRs and IPs to scan.
+            type: list
+            required: false
+            default:
+              - 192.168.0.0/24
+        communities:
+            description: List of communities to query per device.
+            type: list
+            required: false
+            default:
+              - "public"
+        wait:
+          description: wait n milliseconds (1/1000 of a second) between sending packets.
+          type: int
+          default: 10
+          required: false
+        platform:
+          description: Dict containing additional mappings between Platform (key) and regex to match in sysDescr (value).
+          type: dict
+          default: {}
+          required: false
+    notes:
+      - Only discover devices running SNMPv2.
+      - 'TODO: Add caching support.'
+'''
+EXAMPLES = '''
+    # onesixtyone.yaml or any other YAML file-extension
+    plugin: onesixtyone
+    strict: False
+    addresses:
+      - 192.168.0.0/24
+'''
+
+import os
+import re
+import tempfile
+import subprocess
+
+from ansible import constants as C
+from ansible.errors import AnsibleParserError
+from ansible.module_utils._text import to_native
+from ansible.module_utils.common.process import get_bin_path
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+
+try:
+    from netaddr import IPSet
+    HAS_NETADDR = True
+except Exception:
+    HAS_NETADDR = False
+
+try:
+    import gevent
+    from gevent import socket
+    HAS_GEVENT = True
+except Exception:
+    HAS_GEVENT = False
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
```

Comment:
you subclass constructable and cacheable but don't use them?

Reply:
The initial plan was to use them...